packages/framework/src/Form/Admin/Customer/UserFormType.php

```diff
@@ -77,31 +126,41 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 ],
                 'first_options' => [
                     'constraints' => $this->getFirstPasswordConstraints($options['user'] === null),
+                    'label' => t('Password'),
+                ],
+                'second_options' => [
+                    'label' => t('Password again'),
                 ],
                 'invalid_message' => 'Passwords do not match',
             ]);
 
-        if ($options['user'] === null) {
-            $builder
-                ->add('domainId', DomainType::class, [
-                    'required' => true,
-                    'data' => $options['domain_id'],
-                ]);
-            $pricingGroups = $this->pricingGroupFacade->getAll();
-            $groupPricingGroupsBy = 'domainId';
-        } else {
-            $pricingGroups = $this->pricingGroupFacade->getByDomainId($options['domain_id']);
-            $groupPricingGroupsBy = null;
+        if ($user instanceof User) {
+            $builderSystemDataBuilder->add('formId', DisplayOnlyType::class, [
+                'label' => t('ID'),
+                'widget_form_type_data' => $user->getId(),
+            ]);
+
+            $builderSystemDataBuilder->add('domainIcon', DisplayOnlyType::class, [
+                'widget_form_type' => DisplayOnlyDomainIconType::class,
+                'widget_form_type_data' => $user->getDomainId(),
+            ]);
+
+            $builderSystemDataBuilder->add('createdAt', DisplayOnlyType::class, [
+                'label' => t('Date of registration and privacy policy agreement'),
+                'widget_form_type_data' => $user->getCreatedAt()->format(self::DATE_TIME_FORMAT),
+            ]);
+
+            $builderRegisteredCustGroup->add('lastLogin', DisplayOnlyType::class, [
+                'label' => t('Last login'),
+                'widget_form_type' => TextType::class,
```

Comment:
Why not display it directly without the read-only input? It would be consistent with `createdAt`.

Is the `widget_form_type` even necessary? I mean, creating a read-only non-mapped input is not that hard. This is just a one more thing to maintain (imagine needing to pass an option to the inner type or wanting to render its row block instead of just the widget). The `DisplayOnlyType` could be as simple as having a non-mapped `data` option and rendering it in twig. No need for any custom options. KISS.

Reply:
My argument is that there is no need for the `form_type` option.

Reply:
Well ok, my thought was that there will be no need to always insert mapped, required and readonly options. If i want to create display only text type i need to add it there manually, which is wierd for me.